src/utils/dates/dates.ts

```diff
@@ -225,3 +234,41 @@ export function formatDate(
 ): string {
   return format(new Date(date), dateFnsFormat);
 }
+
+/**
+ * Formats a timestamp to a dynamic string based on its proximity to the current date and time.
+ * If the timestamp is from today, it returns the time in 12-hour format.
+ * If the timestamp is from yesterday, it returns the string 'Yesterday'.
+ * If the timestamp is from this week, it returns the day of the week in abbreviated form (e.g. 'Mon').
+ * If the timestamp is from this year, it returns the date in short format (e.g. 'Jan 1').
+ * Otherwise, it returns the date in US format (e.g. '01/01/2022').
+ * @param timestamp The timestamp to format
+ * @returns A string representing the formatted timestamp
+ *
+ * @example
+ * ```typescript
+ * formatTimestampDynamic(new Date());
+ * // => '12:00 AM'
+ *
+ * formatTimestampDynamic('2022-01-01T12:00:00Z');
+ * // => '01/01/2022'
+ * ```
+ */
+export function formatTimestampDynamic(timestamp: Date | string | number) {
+  const date = new Date(timestamp);
+  const today = new Date();
+  console.log(today, sub(today, { weeks: 1 }));
+  if (isToday(date)) {
+    return formatTime12hr(date);
+  }
+  if (isYesterday(date)) {
+    return 'Yesterday';
+  }
+  if (date > sub(today, { weeks: 1 })) {
```

Comment:
better to use `isAfter` from date fns

Reply:
+ this would make dates in future pretty hard to read. do we want that?

Reply:
Can you check once @dineshdevkota001 ?